code/iaas/allocator/src/main/java/io/cattle/platform/allocator/service/AllocatorServiceImpl.java

```diff
@@ -87,13 +87,19 @@ public void mergeLabels(Map<String, String> srcMap, Map<String, String> destMap)
             return;
         }
         for (Map.Entry<String, String> entry : srcMap.entrySet()) {
-            String destValue = destMap.get(entry.getKey());
-            if (StringUtils.isEmpty(destValue)) {
-                destMap.put(entry.getKey(), entry.getValue());
-            } else if (StringUtils.isEmpty(entry.getValue())) {
-                continue;
+            if (entry.getKey().startsWith("io.rancher.scheduler.affinity")) {
+                // merge labels
+                String destValue = destMap.get(entry.getKey());
+                if (StringUtils.isEmpty(destValue)) {
+                    destMap.put(entry.getKey(), entry.getValue());
+                } else if (StringUtils.isEmpty(entry.getValue())) {
+                    continue;
+                } else if (!destValue.contains(entry.getValue())) {
```

Comment:
This is to prevent the same value from constantly appending.  So we don't have something like:
foo=bar,foo=bar,foo=bar,foo=bar,foo=bar